refactor: remove unused code comment in `Proxy.sol`

arthurgousset opened the pull request:
### Description

1. removes unused `TODO` code comments in `Proxy.sol`, which were probably [added by mistake](https://github.com/celo-org/celo-monorepo/pull/10875/files#diff-77393931e4bc5a5ed4b5900b72c2525c39330750268a18d14efeb0b2c242631cR1-R7) in this PR: https://github.com/celo-org/celo-monorepo/pull/10875)

### Other changes

1. manually updates `Proxy.sol` artifacts in `packages/protocol/artifacts/Proxy/` so the `test_verifyArtifacts()` in test in [`ProxyFactory08.t.sol`](packages/protocol/test-sol/unit/common/ProxyFactory08.t.sol) passes before and after the PR (left a note below)
2. simplifes `getCode` argument in `ProxyFactory08.t.sol` for better readability. The change is semantically equivalent see [Foundry docs](https://book.getfoundry.sh/cheatcodes/get-code).

### Other

Leaving a note for future reference:

1. The `proxyInitCode...` file seems to require the bytecode for `Proxy.sol`. I'm not sure if this is the correct way to do it, but I simply copy/pasted the JSON value at `packages/protocol/out/Proxy.sol/Proxy.json` > `bytecode.object.` which is from the Foundry build artifacts.
1. The `proxyBytecode...` file seems to require the deployed bytecode for `Proxy.sol`. I'm not sure if this is the correct way to do it, but I simply copy/pasted the JSON value at `packages/protocol/out/Proxy.sol/Proxy.json` > `deployedBytecode.object.` which is from the Foundry build artifacts.

Unless the bytecodes in these manual artifacts matches the actual Foundry artifacts, the `test_verifyArtifacts()` test in [`ProxyFactory08.t.sol`](packages/protocol/test-sol/unit/common/ProxyFactory08.t.sol) will fail.

I didn't have time to investigate this further or question why manual artifacts are needed in the first place. But, I'm leaving a note here for future reference.

### Tested

Yes, locally and on CI ✅ 

The specific test that should pass before and after this PR is: 

```sh
forge test \
-vvvv \
--match-path "*test-sol/unit/common/ProxyFactory08.t.sol"
```

### Related issues

None, just a small housekeeping PR.

### Backwards compatibility

Yes. 

### Documentation

Yes, in `artifacts/Proxy/README`.

arthurgousset committed:
refactor(Proxy.sol): remove unused code comments

This code change makes a test fail:
test-sol/unit/common/ProxyFactory08.t.sol > test_verifyArtifacts

```sh
$ forge test

Failing tests:
Encountered 1 failing test in test-sol/unit/common/ProxyFactory08.t.sol:ProxyFactoryTest
[FAIL. Reason: panic: assertion failed (0x01)] test_verifyArtifacts() (gas: 188341)
```

I'm not 100% certain but it's maybe because the `packages/protocol/artifacts/Proxy/proxyInitCode0.5.17+commit.d19bba13.hex` artifact is derived from the contract with the code comments.

arthurgousset commented on packages/protocol/contracts/common/Proxy.sol:
This code change makes a test fail:
`test-sol/unit/common/ProxyFactory08.t.sol` > `test_verifyArtifacts`

https://github.com/celo-org/celo-monorepo/blob/dc71c4bc94d53069763a3fca25a53def8c60d48c/packages/protocol/test-sol/unit/common/ProxyFactory08.t.sol#L40-L48

```sh
$ forge test

Failing tests:
Encountered 1 failing test in test-sol/unit/common/ProxyFactory08.t.sol:ProxyFactoryTest
[FAIL. Reason: panic: assertion failed (0x01)] test_verifyArtifacts() (gas: 188341)
```

I'm not 100% certain, but it's maybe because the [`packages/protocol/artifacts/Proxy/proxyInitCode0.5.17+commit.d19bba13.hex`](https://github.com/celo-org/celo-monorepo/blob/519ee36f9e80cb42856a715afa0c6a62c4aad964/packages/protocol/artifacts/Proxy/proxyInitCode0.5.17+commit.d19bba13.hex) artifact is derived from the contract with the code comments. This was added in this PR: 
- https://github.com/celo-org/celo-monorepo/pull/11095

arthurgousset replied:
I imagine a way to fix the failing test is to generate a new artifact from the `Proxy.sol` contract without the code comments. I couldn't find any note on how the hex was generated.

I tried to use the `bytecode` and `deployedBytecode` from the build artifacts, but that doesn't seem to match either. 

```sh
# within the packages/protocol directory
$ forge build
$ jq -r '.bytecode.object' ./out/Proxy.sol/Proxy.json
0x60806040526004361061004a5760003560e01c806303386ba31461015e5780634....
```

```sh
# within the packages/protocol directory
$ forge build
jq -r '.deployedBytecode.object' ./out/Proxy.sol/Proxy.json
0x60806040526004361061004a5760003560e01c806303386ba314...
```

Copy the output into [packages/protocol/artifacts/Proxy/proxyInitCode0.5.17+commit.d19bba13.hex](https://github.com/celo-org/celo-monorepo/blob/519ee36f9e80cb42856a715afa0c6a62c4aad964/packages/protocol/artifacts/Proxy/proxyInitCode0.5.17+commit.d19bba13.hex) and run the tests

```sh
$ forge test
```

That didn't work for me. Perhaps I didn't use the 0.5.17 compiler version? Not sure it's relevant, but it's a difference

https://github.com/celo-org/celo-monorepo/blob/dc71c4bc94d53069763a3fca25a53def8c60d48c/packages/protocol/test-sol/unit/common/ProxyFactory08.t.sol#L41

pahor167 replied:
Did you try to manually remove artifacts and building project again ?

arthurgousset replied:
Yes, I deleted them and tried 

```sh
# top level directory in monorepo
$ yarn build
```

This didn't generate the artifacts. 

I also tried to run the test to see if that generates the artifacts in some way, but it looks like it only reads the files. 

```sh
$ forge test -vvvv \
--match-path "test-sol/unit/common/ProxyFactory08.t.sol"
```

arthurgousset replied:
Yes, I deleted them and tried 

```sh
# top level directory in monorepo
$ yarn build
```

This didn't generate the artifacts. 

I also tried to run the test to see if that generates the artifacts in some way, but it looks like it only reads the files. 

```sh
$ forge test -vvvv \
--match-path "test-sol/unit/common/ProxyFactory08.t.sol"
```

arthurgousset replied:
I asked @martinvol on [Slack](https://clabsco.slack.com/archives/C02LN9FRME1/p1721825388798569?thread_ts=1721748008.256399&cid=C02LN9FRME1):

> I'm not sure how to generate them. Do you remember what you did? I could hack around and try, but curious if you remember what you did.
> I'll add it to the [artifacts/Proxy/Readme.md](https://github.com/celo-org/celo-monorepo/blob/519ee36f9e80cb42856a715afa0c6a62c4aad964/packages/protocol/artifacts/Proxy/Readme.mdREADME) for future reference.

arthurgousset replied:
Response (for when I get back to this): 

> oh I should have left the console.log commented, but you can print the result of `vm.toString(bytecode)` [here](https://github.com/celo-org/celo-monorepo/pull/11095/files#diff-719168f1a039de0564003633325437136f59f6e499301a79ecfd072b1639dad4R55) (edited)

> It is interesting though that a change in comment changed either the bytecode or the init code. I’m sorry you had to go to this rabitthole for a comment change 

> It’s kinda interesting that you get a panic rather than a regular assertion error, I haven’t seen that. I’d try to narrow down what’s the call that fails first to make sure is that.

arthurgousset committed:
Merge branch 'release/core-contracts/12' into arthurgousset/refactor/remove-code-comment-in-proxy-contract

arthurgousset committed:
refactor(ProxyFactory08.t.sol): simplify `getCode` arguments

This is semantically equivalent, see https://book.getfoundry.sh/cheatcodes/get-code

arthurgousset committed:
chore(artifacts/Proxy): adds `bytecode`

I simply copy/pasted the JSON value at `packages/protocol/out/Proxy.sol/Proxy.json` > `bytecode.object.`

arthurgousset committed:
chore(artifacts/Proxy): adds `bytecode`

I simply copy/pasted the JSON value at `packages/protocol/out/Proxy.sol/Proxy.json` > `deployedBytecode.object.`

arthurgousset committed:
docs(artifacts/Proxy/README): adds a note for future reference

arthurgousset committed:
style(artifacts/Proxy/README): fix typo

arthurgousset committed:
style(artifacts/Proxy/README): fix typo

martinvol commented on packages/protocol/contracts/common/Proxy.sol:
```diff
@@ -1,10 +1,7 @@
-// TODO make Iproxy
```
This import is somewhat important and not trivial to implement: This implementation of a proxy has a methods that should be external as public, this make us consume interfaces that are not fully consistent with the implementation. This will make the contract change bytecode, which is not something we can redeploy on mainnet for existing contracts.

arthurgousset replied:
I assume this is a comment you left for future reference. I don't think I can merge this PR without resolving every conversation, so I'll copy/paste this as a comment below for future reference: https://github.com/celo-org/celo-monorepo/pull/11174#issuecomment-2268764007

martinvol commented on packages/protocol/test-sol/unit/common/ProxyFactory08.t.sol:
```diff
@@ -13,7 +13,7 @@ contract ProxyFactoryTest is Test, Utils08 {
 
   function setUp() public {
     proxyFactory08 = new ProxyFactory08();
-    proxyInitCode = vm.getCode(string.concat("out/Proxy.sol/Proxy.json"));
```
this was a bug in foundry that could only use those paths and we probably have it like this elsewhere in the code

arthurgousset replied:
Great point looks like it's used here: 

https://github.com/celo-org/celo-monorepo/blob/67d6acc3268f22807d9f2fc239fd2a2567a18751/packages/protocol/migrations_sol/Migration.s.sol#L131-L133

I propose this gets fixed in a separate PR to separate changes into small PRs.

arthurgousset commented:
Comment from @martinvol for future reference:

> This import is somewhat important and not trivial to implement: This implementation of a proxy has a methods that should be external as public, this make us consume interfaces that are not fully consistent with the implementation. This will make the contract change bytecode, which is not something we can redeploy on mainnet for existing contracts.

Source: https://github.com/celo-org/celo-monorepo/pull/11174#discussion_r1698868045
